azDevOps/azure/templates/steps/deploy-terraform-apply.yml

```diff
@@ -0,0 +1,49 @@
+parameters:
+  environment: ''
+  terraform_artefact_path: ''
+  # Parameters specific to terraform as a map
+  tf_job_input_params: ''
+  tf_job_output_params: ''
+
+steps:
+- bash: pwd; ls -la ; sudo chmod -R 777 ./ ; sudo ls -la; env
+  displayName: 'Environment'
+  workingDirectory: ${{ parameters.terraform_artefact_path }}
+
+- bash: terraform plan -input=false -out=tfplan
+  displayName: 'Terraform: plan'
+  workingDirectory: ${{ parameters.terraform_artefact_path }}
+  env:
```

Comment:
I think we should start naming these templates with prefixes, ie:
deploy-**azure**-terraform-apply.yaml
because some steps will only be valid for azure
Or, break the each step into their own template and create a meta-step coupling the other steps